owtf/webapp/src/containers/TargetsPage/index.js

```diff
@@ -1,19 +1,268 @@
+  handleAlertMsg(alertStyle, alertMsg){
+    this.setState({
+      show: true,
+      alertStyle: alertStyle,
+      alertMsg: alertMsg
+    });
+    setTimeout(() => {
+      this.setState({ show: false });
+    }, 5000);
+  }
+
+  handleTargetUrlsChange({ target }) {
+    this.setState({
+      [target.name]: target.value
+    });
+  }
+
+  isUrl(str) {
+    // TODO: Add all url protocols to support network plugins
+    const urlPattern = /(http|ftp|https):\/\/[\w-]+(\.[\w-]+)+([\w.,@?^=%&amp;:\/~+#-]*[\w@?^=%&amp;\/~+#-])?/;
+    return urlPattern.test(str);
+  }
+
+  // Add new targets using API
+  addNewTargets(targetUrlsString, button) {
+    this.setState({ disabled: true });
+    const lines = this.state.newTargetUrls.split("\n");
+    const targetUrls = [];
+    lines.map(line => {
+      if (this.isUrl(line)) {
+        // Check if a valid url
+        targetUrls.push(line);
+        this.setState({ newTargetUrls: "" });
+      } else if (this.isUrl("http://" + line)) {
+        // If not valid url and adding http:// makes it a valid url
+        targetUrls.push("http://" + line);
+        targetUrls.push("https://" + line);
+        this.setState({ newTargetUrls: "" });
+      } else {
+        this.handleAlertMsg("danger", line + " is not a valid url");
+      }
+    });
+
+    // Since we only have valid urls now in targetUrls, add them using api
+    // Proceed only if there is atleast one valid url
+    if (targetUrls.length > 0) {
+      this.handleAlertMsg("info", "Targets are being added in the background, and will appear in the table soon"); 
+      targetUrls.map(target_url => {
+        this.props.onCreateTarget(target_url);
+        setTimeout(()=> {
+          if(this.props.createError !== false){
+            this.handleAlertMsg("danger", "Unable to add " + unescape(target_url.split("//")[1]));
+          }
+        }, 200);
+      });
+    }
+    this.setState({ disabled: false });
+  }
+
+  componentDidMount() {
+    this.props.onFetchTarget();
+  }
+
+  //return current active session
+  getCurrentSession() {
+    return this.refs.sessionReference.getWrappedInstance().getCurrentSession()
+  }
+
+  //download list of targets as txt file
+  exportTargets() {
+    const targetsArray = [];
+    this.props.targets.map((target) => {
+      targetsArray.push(target.target_url+'\n');
+    });
+    const element = document.createElement("a");
+    const file = new Blob(targetsArray, {type: 'text/plain;charset=utf-8'});
+    element.href = URL.createObjectURL(file);
+    element.download = "targets.txt";
+    element.click();
   }
 
   render() {
+    const { targets, fetchLoading, fetchError } = this.props;
+    const TargetsTableProps = {
+      targets: targets,
+      getCurrentSession: this.getCurrentSession,
+      handleAlertMsg: this.handleAlertMsg,
+    }
     return (
-      <Sessions />
+      <Grid>
+        <Row>
+          <Col>{this.renderAlert()}</Col>
+        </Row>
+        <Row>
+          <Breadcrumb>
```

Comment:
@sharmamohit123 Try using react-router to navigate across breadcrumbs. You can google for current implementations.